modules/hitlnext/src/views/full/app/components/HandoffListHeader.tsx

```diff
@@ -89,11 +99,62 @@ const HandoffListHeader: FC<Props> = ({ filterOptions, sortOption, setFilterOpti
     }
   ]
 
+  const renderButtons = (tags: string[]): ToolbarButtonProps[] => {
+    const tagElements: ToolbarButtonProps[] = tags.map(tag => {
+      return {
+        content: (
+          <Checkbox
+            checked={filterOptions.tags.includes(tag)}
+            label={tag}
+            onChange={() =>
+              setFilterOptions({
+                ...filterOptions,
+                tags:
+                  filterOptions.tags && filterOptions.tags.includes(tag)
+                    ? _.pull(filterOptions.tags, tag)
+                    : [...filterOptions.tags, tag]
+              })
+            }
+          />
+        )
+      }
+    })
+    return [
+      {
+        icon: 'sort',
+        optionsItems: [
+          {
+            label: lang.tr('module.humanhelpnext.sort.mostRecentlyCreated'),
+            selected: sortOption === 'mostRecent',
+            action: () => {
+              setSortOption('mostRecent')
+            }
+          },
+          {
+            label: lang.tr('module.hitlnext.sort.leastRecentlyCreated'),
+            selected: sortOption === 'leastRecent',
+            action: () => {
+              setSortOption('leastRecent')
+            }
+          }
+        ],
+        tooltip: lang.tr('module.hitlnext.sortBy'),
+        disabled
+      },
+      {
+        icon: 'filter',
+        optionsItems: buttons[1].optionsItems.concat(tagElements),
+        tooltip: lang.tr('module.hitlnext.filterBy'),
+        disabled
+      }
+    ]
+  }
+
   return (
     <MainLayout.Toolbar
       className={style.hitlToolBar}
       tabs={[{ id: 'handoffs', title: lang.tr('module.hitlnext.sidebar.tab') }]}
-      buttons={buttons}
+      buttons={tags && tags.length > 0 ? renderButtons(tags) : buttons}
```

Comment:
Why not move this logic in your renderButtons function ? This way we would call renderButtons in any case

Reply:
I agree, I will make the changes.